starknet_py/utils/tuple_dataclass.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+from dataclasses import dataclass, fields, make_dataclass
+from typing import Dict, Optional, Tuple
+
+
+@dataclass(frozen=True)
+class TupleDataclass:
+    """
+    Dataclass that behaves like a tuple at the same time.
+    """
+
+    # getattr is called when attribute is not found in object
+    # This way pyright will know that there might be some arguments it doesn't know about and will stop complaining
+    # about some fields that don't exist statically.
+    def __getattr__(self, item):
+        # This should always fail - only attributes that don't exist end up in here.
+        # We use __getattribute__ to get the native error.
+        return super().__getattribute__(item)
```

Comment:
Will it ever be called? From what I understand from docs:

Called when the default attribute access fails with an [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError) (either [__getattribute__()](https://docs.python.org/3/reference/datamodel.html#object.__getattribute__) raises an [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError) because name is not an instance attribute or an attribute in the class tree for self; or [__get__()](https://docs.python.org/3/reference/datamodel.html#object.__get__) of a name property raises [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError)). This method should either return the (computed) attribute value or raise an [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError) exception.

It will only get called if two other methods mentioned raise AttributeError. We don't implement any of these, only `__getitem__` which raises (at least should) IndexError.

Reply:
It will be called in case when someone wants to use an attribute that object doesn't have. Like this:
```
result = TupleDataclass.from_dict(input_dict)
result.unknown_property
```

Reply:
Okay it makes sense. Perhaps info about `.` operator falling back to this could be included in the comment? It's kind of there but also it isn't.